From 2fadc3f547b9b3c23514d0cf744226b7518f59ea Mon Sep 17 00:00:00 2001
From: Kai-Chieh Ku <kjackie@gmail.com>
Date: Wed, 16 Oct 2013 22:02:33 +0800
Subject: [PATCH] TA release build has bug!

---
 ta/ta.h | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/ta/ta.h b/ta/ta.h
index 1a9738a..2398e79 100644
--- a/ta/ta.h
+++ b/ta/ta.h
@@ -140,7 +140,13 @@ void *ta_oom_p(void *p);
 #define ta_oom_g(ptr) (TA_TYPEOF(ptr))ta_oom_p(ptr)
 
 void ta_enable_leak_report(void);
+
+#ifdef TA_MEMORY_DEBUGGING
 void *ta_dbg_set_loc(void *ptr, const char *name);
 void *ta_dbg_mark_as_string(void *ptr);
+#else
+static inline void *ta_dbg_set_loc(void *ptr, const char *name) { return ptr; }
+static inline void *ta_dbg_mark_as_string(void *ptr) { return ptr; }
+#endif
 
 #endif
-- 
1.8.1.5

